examples/daal/cpp/source/decision_forest/df_cls_default_dense_extratrees_batch.cpp

```diff
@@ -0,0 +1,154 @@
+!******************************************************************************/
+
+/**
+ * <a name="DAAL-EXAMPLE-CPP-DF_CLS_DEFAULT_DENSE_EXTRATREES_BATCH"></a>
+ * \example df_cls_default_dense_extratrees_batch.cpp
+ */
+
+#include "daal.h"
+#include "service.h"
+
+using namespace daal;
+using namespace daal::algorithms;
+using namespace daal::data_management;
+using namespace daal::algorithms::decision_forest::classification;
+
+/* Input data set parameters */
+const std::string trainDatasetFileName = "../data/batch/df_classification_train.csv";
+const std::string testDatasetFileName = "../data/batch/df_classification_test.csv";
+const size_t categoricalFeaturesIndices[] = { 2 };
+const size_t nFeatures = 3; /* Number of features in training and testing data sets */
+
+/* Decision forest parameters */
+const size_t nTrees = 10;
+const size_t minObservationsInLeafNode = 8;
+const size_t minObservationsInSplitNode = 16;
+const double minWeightFractionInLeafNode = 0.0; /* It must be in segment [0.0, 0.5] */
+const double minImpurityDecreaseInSplitNode = 0.0; /* It must be greater than or equal to 0.0 */
+
+const size_t nClasses = 5; /* Number of classes */
+
+training::ResultPtr trainModel();
+void testModel(const training::ResultPtr& res);
+void loadData(const std::string& fileName, NumericTablePtr& pData, NumericTablePtr& pDependentVar);
+
+int main(int argc, char* argv[]) {
+    checkArguments(argc, argv, 2, &trainDatasetFileName, &testDatasetFileName);
+
+    training::ResultPtr trainingResult = trainModel();
+    testModel(trainingResult);
+
+    return 0;
+}
+
+training::ResultPtr trainModel() {
+    /* Create Numeric Tables for training data and dependent variables */
+    NumericTablePtr trainData;
+    NumericTablePtr trainDependentVariable;
+
+    loadData(trainDatasetFileName, trainData, trainDependentVariable);
+
+    /* Create an algorithm object to train the decision forest classification model */
+    training::Batch<float, training::defaultDense> algorithm(nClasses);
+
+    /* Pass a training data set and dependent values to the algorithm */
+    algorithm.input.set(classifier::training::data, trainData);
+    algorithm.input.set(classifier::training::labels, trainDependentVariable);
+
+    algorithm.parameter().nTrees = nTrees;
+    algorithm.parameter().featuresPerNode = nFeatures;
+    algorithm.parameter().minObservationsInLeafNode = minObservationsInLeafNode;
+    algorithm.parameter().minObservationsInSplitNode = minObservationsInSplitNode;
+    algorithm.parameter().minWeightFractionInLeafNode = minWeightFractionInLeafNode;
+    algorithm.parameter().minImpurityDecreaseInSplitNode = minImpurityDecreaseInSplitNode;
+    algorithm.parameter().varImportance = algorithms::decision_forest::training::MDI;
+    /* Enable ExtraTrees classification algorithm with bootstrap=false and random splitter*/
+    algorithm.parameter().splitter = algorithms::decision_forest::training::random;
```

Comment:
I don't think it's meaningful to create separate decision forest examples if the only difference is in this row.
Might be better to set `splitter` as argument in `trainModel` function and run both splitters in same file